src/api/db/data/20181214100207_remove_duplicated_flags.rb

```diff
@@ -0,0 +1,24 @@
+class RemoveDuplicatedFlags < ActiveRecord::Migration[5.2]
+  def up
+    attributes = [:flag, :repo, :architecture_id, :package_id, :project_id]
+    # whenever 2 flags are for the same 'thing' (in _meta), we remove all but one
+    last_flags = Flag.group([:status] + attributes).select('MAX(id) as id')
+    # we need to pluck the ids (even though it's a lot) as you can't delete
+    # from within group queries
+    Flag.where.not(id: last_flags).in_batches do |batch|
+      Flag.where(id: batch.pluck(:id)).delete_all
```

Comment:
Instead of running the query twice I would do this:
```ruby
Flag.where.not(id: last_flags).in_batches.delete_all
```

Reply:
did you read the comment?

Reply:
But calling `last_flags` within the `Flag.where...` should execute the group query, or not? And if rails chains the two queries together you probably could prevent this by triggering the "group" query via `last_flags.pluck(:id)`.

Reply:
can you imagine how many last_flags we have and how large that plucked array would be? There is a reason this is done in batches, because it's *a lot*

Reply:
Okay Okay, you convinced me:wink: